javascripts/discourse/components/featured-homepage-topics.js

```diff
@@ -1,109 +1,68 @@
-        .findFiltered("topicList", {
-          filter: "latest",
-          params: {
-            tags: [`${settings.featured_tag}`],
-            order: sortOrder,
-          },
-        })
-        .then((topicList) => {
-          let featuredTagTopics = [];
+export default class MyComponent extends Component {
+  @service router;
+  @service store;
+  @service siteSettings;
+  @tracked titleElement = null;
+  @tracked featuredTagTopics = null;
 
-          topicList.topics.forEach((topic) =>
-            topic.image_url ? featuredTagTopics.push(Topic.create(topic)) : ""
-          );
+  constructor() {
+    super(...arguments);
+    this.router.on("routeDidChange", this.checkShowHere);
+  }
 
-          this.set(
-            "featuredTagTopics",
-            featuredTagTopics.slice(0, settings.number_of_topics)
-          );
-        });
-    }
-  },
+  willDestroy() {
+    this.router.off("routeDidChange", this.checkShowHere);
+  }
 
-  _checkClass() {
+  @action
+  checkShowHere() {
     if (!this.showHere) {
       document.querySelector("body").classList.remove(FEATURED_CLASS);
+    } else {
+      document.querySelector("body").classList.add(FEATURED_CLASS);
     }
-  },
+  }
 
-  init() {
-    this._super(...arguments);
-    this._getBanner();
-    this._checkClass();
-  },
-
-  didInsertElement() {
-    this.appEvents.on("page:changed", this, "_checkClass");
-  },
-
-  willDestroyElement() {
-    this.appEvents.off("page:changed", this, "_checkClass");
-    document.querySelector("body").classList.remove(FEATURED_CLASS);
-  },
-
-  @discourseComputed("router.currentRoute", "router.currentRouteName")
-  showHere(currentRoute, currentRouteName) {
-    let showHere;
+  get showHere() {
+    let currentRoute = this.router.currentRoute;
+    let currentRouteName = this.router.currentRouteName;
 
     if (currentRoute) {
-      if (settings.show_on === "homepage") {
-        showHere = currentRouteName === `discovery.${defaultHomepage()}`;
-      } else if (settings.show_on === "top_menu") {
-        const topMenuRoutes = this.siteSettings.top_menu
-          .split("|")
-          .filter(Boolean);
-        showHere = topMenuRoutes.includes(currentRoute.localName);
-      } else if (settings.show_on === "all") {
-        showHere =
-          currentRouteName.indexOf("editCategory") &&
-          currentRouteName.indexOf("admin") &&
-          currentRouteName.indexOf("full-page-search");
-      } else {
-        return false;
+      switch (settings.show_on) {
+        case "homepage":
+          return currentRouteName === `discovery.${defaultHomepage()}`;
+        case "top_menu":
+          const topMenuRoutes = this.siteSettings.top_menu
+            .split("|")
+            .filter(Boolean);
+          return topMenuRoutes.includes(currentRoute.localName);
+        case "all":
+          return !["editCategory", "admin", "full-page-search"].some((route) =>
+            currentRouteName.includes(route)
+          );
```

Comment:
⛳  and a (tiny) perf improvement:

```suggestion
          return !/editCategory|admin|full-page-search/.test(currentRouteName);
```